light/client.go

```diff
@@ -258,6 +269,24 @@ func NewClientFromTrustedStore(
 	return c, nil
 }
 
+// isBlacklisted checks whether provider is black listed
+// NOTE: requires a providerMutex lock
+func (c *Client) isBlacklisted(p provider.Provider) bool {
+	timestamp, exists := c.blacklist[p.ID()]
+	if !exists {
+		return false
+	}
+
+	// If the provider is found, check the TTL
+	if time.Since(timestamp) > c.blacklistTTL {
```

Comment:
This is fine for me, another option is to use https://github.com/hashicorp/golang-lru, which manage the expiration automatically for you

Reply:
Yea will leave as is for now. Can switch to this if we want in another PR.